### PR TITLE
Enabled `tetra` bash autocompletion in the Tetragon image

### PR DESCRIPTION
**Pull request by PhilipSchmid**
### Description
Minor UX improvement when troubleshooting inside the Tetragon container image:

```bash
77a8a8762616:/# tetra    # pressing <tab><tab>
bugtool          (Produce a tar archive with debug information)                probe            (Probe for eBPF system features availability)
completion       (Generate the autocompletion script for the specified shell)  stacktrace-tree  (Manage stacktrace trees)
cri              (connect to CRI)                                              status           (Print health status)
getevents        (Print events)                                                tracingpolicy    (Manage tracing policies)
help             (Help about any command)                                      version          (Print version from CLI and server)
loglevel         (Get and dynamically change the log level)
```

### TODOs
- [x] Check if there's a better way to install `bash-completion` in a separate stage and only copying over the relevant files. Why? See outputs of `grype`👇 

Without `bash-completion` installed:
```bash
 ✔ Vulnerability DB                [no update available]
 ✔ Loaded image                                                                                                                                                                 af7eb1d3b2a6
 ✔ Parsed image                                                                                                      sha256:af7eb1d3b2a6396945a08f836e6aaca937027c985cd07e4bf6f3bccb216d50a8
 ✔ Cataloged contents                                                                                                       1bebc98c8d0b81c46ab1f3e5ccda178d7b838a9c291a445f32c650c333291194
   ├── ✔ Packages                        [310 packages]
   ├── ✔ File digests                    [188 files]
   ├── ✔ File metadata                   [188 locations]
   └── ✔ Executables                     [78 executables]
 ✔ Scanned for vulnerabilities     [0 vulnerability matches]
   ├── by severity: 0 critical, 0 high, 0 medium, 0 low, 0 negligible
   └── by status:   0 fixed, 0 not-fixed, 0 ignored
No vulnerabilities found
```

With `bash-completion` installed:
```bash
 ✔ Vulnerability DB                [no update available]
 ✔ Loaded image                                                                                                                                                                 dd56dbaea46f
 ✔ Parsed image                                                                                                      sha256:dd56dbaea46f5212cbc49f3f68d023f8084f9c05f894be49828cbebec7da0830
 ✔ Cataloged contents                                                                                                       eb044ca6d3b763ff5a1e187ea2725791a08bb6f183e5b8ad8fecabaa83166846
   ├── ✔ Packages                        [312 packages]
   ├── ✔ File digests                    [696 files]
   ├── ✔ File metadata                   [696 locations]
   └── ✔ Executables                     [78 executables]
 ✔ Scanned for vulnerabilities     [0 vulnerability matches]
   ├── by severity: 0 critical, 0 high, 0 medium, 0 low, 0 negligible
   └── by status:   0 fixed, 0 not-fixed, 0 ignored
No vulnerabilities found
```

`188 files` vs. `696 files` and `188 locations` vs. `696 locations`

### Release Note
```release-note
Enabled `tetra` bash autocompletion in the Tetragon image
```

### CODE REVIEW

**Review by kkourt — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -80,10 +80,18 @@ ARG TARGETARCH
 ARG BPFTOOL_TAG=v7.2.0-snapshot.0
 RUN curl -L https://github.com/libbpf/bpftool/releases/download/${BPFTOOL_TAG}/bpftool-${BPFTOOL_TAG}-${TARGETARCH}.tar.gz | tar xz && chmod +x bpftool
 
+# Get bash-completion manifests and generate tetra CLI bash
+# autocompletions (we don't want all bash-completions in the base-build)
+FROM docker.io/library/alpine:3.20.3@sha256:beefdbd8a1da6d2915566fde36db9db0b524eb737fc57cd1367effd16dc0d06d AS cli-autocomplete
+COPY --from=tetragon-builder /go/src/github.com/cilium/tetragon/tetra /usr/bin/
+RUN apk add --no-cache bash-completion && \
+    tetra completion bash | tee /etc/bash_completion.d/tetra > /dev/null && \
```
Comment (mtardy) [inline Dockerfile:88]: nit: maybe you don't need tee if you redirect stdout to dev null?
```suggestion
    tetra completion bash > /etc/bash_completion.d/tetra && \
```

**Review by mtardy — APPROVED** [review]
Looks good, just a small comment to look before merging this

**Comment by PhilipSchmid** [comment]
> Due to improvements in the code handling DNS in version 1.16, this issue is no longer present.

See ZD ticket for further details.

File: Dockerfile (hunk 1 above)
Comment (PhilipSchmid) [reply Dockerfile]: Right, good point! Thanks. I committed & squashed the suggestion.

**Comment by mtardy** [comment]
did you want to squash 982dc75de7c667a5f720a7849287cd581f3d2126 and 66f171d52794d151a33e2ee678d456b7aeaad494 or we merge as is?

**Comment by PhilipSchmid** [comment]
@mtardy Sure, thanks! I just squshed them.

**Pull request merged**